modules/ximgproc/src/thinning.cpp

```diff
@@ -69,6 +69,7 @@ static void thinningIteration(Mat img, int iter, int thinningType){
 // Apply the thinning procedure to a given image
 void thinning(InputArray input, OutputArray output, int thinningType){
     Mat processed = input.getMat().clone();
+    CV_Assert(processed.type == CV_8UC1);
```

Comment:
Consider using `CV_CheckTypeEQ(input.type(), CV_8UC1, "");` for clear diagnostic messages.